File: crates/algosketch-core/tests/fixtures/quick_sort.java
class Algorithms {
    int[] quick_sort(int[] items, int low, int high) {
        if (low < high) {
            int pivot = partition(items, low, high);
            quick_sort(items, low, pivot - 1);
            quick_sort(items, pivot + 1, high);
        }
        return items;
    }

    int partition(int[] items, int low, int high) {
        int pivot = items[high];
        int i = low;
        for (int j = low; j < high; j = j + 1) {
            if (items[j] < pivot) {
                int temp = items[i];
                items[i] = items[j];
                items[j] = temp;
                i = i + 1;
            }
        }
        int temp = items[i];
        items[i] = items[high];
        items[high] = temp;
        return i;
    }
}

File: crates/algosketch-core/tests/fixtures/quick_sort.go
package main

func quick_sort(items []int, low int, high int) []int {
	if low < high {
		pivot := partition(items, low, high)
		quick_sort(items, low, pivot-1)
		quick_sort(items, pivot+1, high)
	}
	return items
}

func partition(items []int, low int, high int) int {
	pivot := items[high]
	i := low
	for j := low; j < high; j++ {
		if items[j] < pivot {
			temp := items[i]
			items[i] = items[j]
			items[j] = temp
			i = i + 1
		}
	}
	temp := items[i]
	items[i] = items[high]
	items[high] = temp
	return i
}
